src/proj7BittingCerratoCohenEllmer/bantam/lexer/Scanner.java

```diff
@@ -186,26 +152,34 @@ private boolean isCompleteComment(Stack<Character> spellingStack) {
         char secondChar = spellingStack.get(1);
         char lastChar = spellingStack.peek();
         if (secondChar == '/') {
-            return lastChar == '\n' || lastChar == '\r';
+            if (lastChar == '\n' || lastChar == '\r') {
+                skippedLastToken = '\0';
+                return true;
+            }
         } else if (secondChar == '*') {
-            char secondToLastChar = spellingStack.get(spellingStack.size() - 1);
-            return secondToLastChar == '*' && lastChar == '/';
+            char secondToLastChar = spellingStack.get(spellingStack.size() - 2);
+            if (secondToLastChar == '*' && lastChar == '/') {
+                skippedLastToken = '\0';
+                return true;
+            }
         }
         return false;
     }
 
     private boolean isCompleteMath(Stack<Character> spellingStack) {
         char lastChar = spellingStack.peek();
-        if (lastChar == ' ' || lastChar == '=') {
-            // set token type
+        if (Character.isWhitespace(lastChar)) {
+            skippedLastToken = spellingStack.pop();
+            return true;
+        } else if (lastChar == '=') {
```

Comment:
All the branches that just reset skippedLastToken and return true could be combined